server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSConverseOptionsCheckUtility.java

```diff
@@ -108,6 +109,13 @@ private void checkRule(CICSParser.Cics_converse_groupContext ctx) {
       checkHasIllegalOptions(ctx.cics_converse_erase(), "ERASE");
     }
 
+    if (!ctx.cics_maxlength().isEmpty())
+      checkHasMandatoryOptions(ctx.cics_converse_tolength(), ctx, "TOLENGTH OR TOFLENGTH");
+
+    if (noLengthOptionsEnabled()) {
+      checkHasMandatoryOptions(ctx.cics_converse_fromlength(), ctx, "FROMLENGTH");
+      checkHasMandatoryOptions(ctx.cics_converse_tolength(), ctx, "TOLENGTH OR TOFLENGTH");
+    }
```

Comment:
to avoid diagnosing the same problem twice.
```suggestion
    if (noLengthOptionsEnabled() || !ctx.cics_maxlength().isEmpty())
      checkHasMandatoryOptions(ctx.cics_converse_tolength(), ctx, "TOLENGTH OR TOFLENGTH");

    if (noLengthOptionsEnabled())
      checkHasMandatoryOptions(ctx.cics_converse_fromlength(), ctx, "FROMLENGTH");
```